cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/co/cask/cdap/etl/proto/v2/ETLBatchConfig.java

```diff
@@ -36,21 +38,25 @@
   private final Engine engine;
   private final String schedule;
   private final Resources driverResources;
-  private final Set<ETLStage> actions;
+  private final List<ETLStage> postActions;
+  // for backwards compatibility
+  private final List<ETLStage> actions;
 
   private ETLBatchConfig(Set<ETLStage> stages,
                          Set<Connection> connections,
-                         Set<ETLStage> actions,
+                         List<ETLStage> postActions,
                          Resources resources,
                          boolean stageLoggingEnabled,
                          Engine engine,
                          String schedule,
                          Resources driverResources) {
     super(stages, connections, resources, stageLoggingEnabled);
-    this.actions = ImmutableSet.copyOf(actions);
+    this.postActions = ImmutableList.copyOf(postActions);
     this.engine = engine;
     this.schedule = schedule;
     this.driverResources = driverResources;
+    // field only exists for backwards compatibility -- used by convertOldConfig()
+    this.actions = null;
```

Comment:
We can't make them completely separate because we have to parameterize the application with an object (AbstractApplication<ETLBatchConfig>), so in order to make the app support both v1 and v2, the v2 object needs to be backwards compatible.